compiler/rustc_parse/src/parser/nonterminal.rs

```diff
@@ -62,7 +58,7 @@ impl<'a> Parser<'a> {
                 },
                 _ => false,
             },
-            NonterminalKind::Pat => match token.kind {
+            pat_kind @ (NonterminalKind::Pat2018 | NonterminalKind::Pat2021) => match token.kind {
```

Comment:
```suggestion
            NonterminalKind::Pat2018 | NonterminalKind::Pat2021 => match token.kind {
```
`kind` is already available.